Client/src/components/navbar.tsx

```diff
@@ -0,0 +1,186 @@
+            {location.pathname === '/' && (
+              <button className="button" onClick={() => navigate('/about')}>
+                ABOUT
+              </button>
+            )}
+            {isAuthenticated && !(location.pathname === '/profile') && (
+              <button className="button" onClick={() => navigate('/profile')}>
+                PROFILE
+              </button>
+            )}
+            {!isLoading && !user && (
+              <button className="button" onClick={() => loginWithRedirect()}>
+                LOGIN
+              </button>
+            )}
+            {!isLoading && user && (
+              <button className="button" onClick={() => logout()}>
+                LOGOUT
+              </button>
+            )}
+          </MediaQuery>
+          <MediaQuery maxWidth={950}>
+            <div
+              className="container"
+              onClick={() => {
+                toggleNavbar();
+                toggleModal();
+              }}
+            >
+              <div className={`bar1 ${toggle && 'active'}`} />
+              <div className={`bar2 ${toggle && 'active'}`} />
+              <div className={`bar3 ${toggle && 'active'}`} />
+            </div>
+          </MediaQuery>
+        </div>
+        <MediaQuery maxWidth={950}>
+          <div className={`modal-body open-${modalOpen}`}>
+            {location.pathname === '/' && (
+              <button
+                className="button"
+                onClick={() => {
+                  navigate('/about');
+                  toggleModal();
+                  toggleNavbar();
+                }}
+              >
+                ABOUT
+              </button>
+            )}
+            {(location.pathname === '/game' || location.pathname === '/profile') && user && (
+              <button
+                className="button"
+                onClick={() => {
+                  navigate('/learning');
+                  toggleModal();
+                  toggleNavbar();
+                }}
+              >
+                LEARN
+              </button>
+            )}
+              {(location.pathname === '/learning' || location.pathname === '/profile') && user && (
+              <button
+                className="button"
+                onClick={() => {
+                  navigate('/game');
+                  toggleModal();
+                  toggleNavbar();
+                }}
+              >
+                GAME
+              </button>
+            )}
+            {!(location.pathname === '/profile') && user && (
+              <button
+                className="button"
+                onClick={() => {
+                  navigate('/profile');
+                  toggleModal();
+                  toggleNavbar();
+                }}
+              >
+                PROFILE
+              </button>
+            )}
+             {!user && (
+              <button
+                className="button"
+                onClick={() => {
```

Comment:
the trigger on checking if the user is authenticated or loading can be done on a separate Component -NEW TASK

Reply:
Changed